### PR TITLE
Refactor projection actor before adding new functionality

### PR DESCRIPTION
**Pull request by klochowicz**
Create watch channels within the actor, returning a single struct called `Feeds`.
Manage the struct with Rocket and use it in SSE events.

Changing the watch channel types etc. will be much easier after this cleanup.

### CODE REVIEW

**Comment by klochowicz** [comment]
First part of #697 (actual changes will be built on top of this PR)

File: daemon/src/projection.rs [hunk 1]
```diff
@@ -1,55 +1,82 @@
...
-            tx_order,
-            tx_quote,
-            tx_settlements,
-            tx_connected_takers,
-        }
+    pub fn new(init_cfds: Vec<Cfd>, init_quote: Quote) -> (Self, Feeds) {
+        let (tx_cfds, rx_cfds) = watch::channel(init_cfds);
+        let (tx_order, rx_order) = watch::channel::<Option<Order>>(None);
+        let (tx_update_cfd_feed, rx_update_cfd_feed) =
+            watch::channel::<UpdateCfdProposals>(HashMap::new());
```
Comment (thomaseizinger) [inline daemon/src/projection.rs]: Don't think you should need turbo fish here if you are directly assigning to fields.

**Review by thomaseizinger — APPROVED** [review]

File: daemon/src/projection.rs (hunk 1 above)
Comment (klochowicz) [reply daemon/src/projection.rs]: indeed! changed :)

**Pull request merged**